students/Briadean/lesson09/OOMailroom.py

```diff
@@ -0,0 +1,177 @@
+import sys
+
+donors_dict = {"Retsuko Rarecho": [100.00, 25.00, 75.00, 200.00],
+               "Fenneko Inoue": [10.00, 200.00, 75.00],
+               "Haida Kato": [15.00, 15.00, 15.00, 15.00, 15.00],
+               "Gori Tsuruta": [25.00, 30.00, 60.00, 90.00],
+               "Washimi Koiwasaki": [10.00, 10.00, 10.00, 35.00, 15.00]}
+
+
+class Donor:
+    def __init__(self, name, donations=None):
+        self.name = name.title()
+        self.donations = [] if donations is None else donations
+
+    def add_donation(self, donation):
+        self.donations.append(donation)
+
+
+class DonorDB:
+    def __init__(self):
+        self._donors = {}
+
+    def add_donor(self, donor):
+        #  fixme add a check to make sure im not overwriting anything already in the db
+        self._donors[donor.name] = donor
+
+    def print_donors(self):
+        print("Donor List:")
+        for donor in self._donors:
+            print(donor)
+
+    def generate_report(self):
+        """Build a report of donors"""
+        report_lines = []
+        for donor in self._donors.values():
+            total_given = sum(donor.donations)
+            num_gifts = len(donor.donations)
```

Comment:
It would be better to define methods for these total_given and num_gifts in your Donor class...